### PR TITLE
Update some dialogs

### PR DESCRIPTION
**Pull request by hdinia**

### CODE REVIEW

File: webapp/src/components/App/Studies/CreateStudyDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    }
+  };
+
+  const publicModeList: Array<GenericInfo> = [
+    { id: "NONE", name: t("study.nonePublicMode") },
+    { id: "READ", name: t("study.readPublicMode") },
+    { id: "EXECUTE", name: t("study.executePublicMode") },
+    { id: "EDIT", name: t("global.edit") },
+    { id: "FULL", name: t("study.fullPublicMode") },
+  ];
```
Comment (pl-buiquang) [inline webapp/src/components/App/Studies/CreateStudyDialog.tsx]: put this in common/utils/const.ts and remove the translation, instead translate key when using this object).
refacto also PropertiesDialog and NavHeader to use the new const object

File: webapp/src/components/App/Studies/CreateStudyDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+function CreateStudyDialog(props: Props) {
+  const [t] = useTranslation();
+  const { open, onClose } = props;
+  const { enqueueSnackbar } = useSnackbar();
+  const enqueueErrorSnackbar = useEnqueueErrorSnackbar();
+  const versionList = useAppSelector(getStudyVersionsFormatted);
+  const mounted = usePromise();
+  const dispatch = useAppDispatch();
+
+  const groupList = useAppSelector(getGroups);
```
Comment (pl-buiquang) [inline webapp/src/components/App/Studies/CreateStudyDialog.tsx]: the groups in the store are wrong.
Those stored are the one with details (containing the list of users associated with the group) which are limited to those which the user is admin of (or superadmin).
Previous `getGroups()` call listed all the groups the user is in.
Either make two calls to merge the two lists (with and without details) in the store or create a separate store.
Note that the `getGroups` with detail is only needed when the user go to the settings page (where he needs to have the list of users associated with groups so to modifiy them if admin)

**Review by pl-buiquang — CHANGES_REQUESTED** [review]

File: webapp/src/components/App/Studies/CreateStudyDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+  const handleSubmit = async (data: SubmitHandlerPlus<FieldValues>) => {
```
Comment (skamril) [inline webapp/src/components/App/Studies/CreateStudyDialog.tsx:50]: Add comment blocks

File: webapp/src/components/App/Studies/CreateStudyDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+          e as AxiosError
+        );
+      }
+      onClose();
+    } else {
+      enqueueSnackbar(t("global.error.emptyName"), { variant: "error" });
+    }
+  };
+
+  const publicModeList: Array<GenericInfo> = [
```
Comment (skamril) [inline webapp/src/components/App/Studies/CreateStudyDialog.tsx]: Duplicate with other components

**Review by skamril — CHANGES_REQUESTED** [review]

File: webapp/src/components/common/utils/contants.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { GenericInfo } from "../../../common/types";
+
```
Comment (pl-buiquang) [inline webapp/src/components/common/utils/contants.ts]: rename "constants.ts" (typo)

**Review by pl-buiquang — APPROVED** [review]

File: webapp/src/components/App/Singlestudy/NavHeader.tsx [hunk 6]
```diff
@@ -102,24 +103,17 @@ function NavHeader(props: Props) {
...
-      { id: "NONE", name: t("study.nonePublicModeText") },
-      { id: "READ", name: t("study.readPublicModeText") },
-      { id: "EXECUTE", name: t("study.executePublicModeText") },
-      { id: "EDIT", name: t("study.editPublicModeText") },
-      { id: "FULL", name: t("study.fullPublicModeText") },
-    ],
-    [t]
-  );
-
   const getPublicModeLabel = useMemo((): string => {
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/NavHeader.tsx]: `useMemo` not necessary.

```javascript
const publicModelLabel = PUBLIC_MODE_LIST.find((mode) => mode.id === study?.publicMode)?.name || "";
```

File: webapp/src/components/App/Singlestudy/PropertiesDialog/index.tsx [hunk 7]
```diff
@@ -70,7 +71,46 @@ function PropertiesModal(props: Props) {
     return tpmTagsChanged;
   }, [initTags, tags]);
 
-  const onSubmit = async () => {
+  usePromiseWithSnackbarError(
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/PropertiesDialog/index.tsx]: You can remove `const [groupList, setGroupList] = useState<Array<GroupDTO>>([]);`

```javascript
const { data: groupList = [] } = usePromiseWithSnackbarError(
    getGroups, { errorMessage: t("settings.error.groupsError") }
);
```

**Review by skamril — CHANGES_REQUESTED** [review]

**Review by skamril — APPROVED** [review]

**Pull request merged**